src/Tasks/sdk-tasks/ExtractArchiveToDirectory.cs

```diff
@@ -185,7 +185,7 @@ private string GetFullDirectoryPathWithSeperator(string directory)
         {
             string fullDirectoryPath = Path.GetFullPath(directory);
 
-            if (!fullDirectoryPath.EndsWith(Path.DirectorySeparatorChar.ToString()))
+            if (!fullDirectoryPath.EndsWith(Path.DirectorySeparatorChar.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
What does this actually do? I wouldn't expect a DirectorySeparatorChar to care about case.

Reply:
And if it actually does something very subtle, should we make this OrdinalIgnoreCase for windows and Ordinal for linux?

Reply:
It's required for these sorts of checks because a malicious user could actually pass in the code for the directoryseparator and we'd match. Feels like the library should handle this better but that's how it works today.
Console.WriteLine("c:\\test\\\u200d".EndsWith("\\")); // TRUE